app/src/main/java/com/vero/tasky/agenda/presentation/taskdetails/TaskDetailsViewModel.kt

```diff
@@ -0,0 +1,144 @@
+package com.vero.tasky.agenda.presentation.taskdetails
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.vero.tasky.agenda.domain.model.ModificationType
+import com.vero.tasky.agenda.domain.usecase.task.TaskUseCases
+import com.vero.tasky.agenda.presentation.model.ReminderRange
+import com.vero.tasky.agenda.presentation.util.LocalDateParser
+import com.vero.tasky.core.presentation.navigation.NavigationConstants
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+import java.time.LocalDate
+import java.time.LocalTime
+import javax.inject.Inject
+
+@HiltViewModel
+class TaskDetailsViewModel @Inject constructor(
+    private val taskUseCases: TaskUseCases,
+    private val savedStateHandle: SavedStateHandle
+) : ViewModel() {
+
+    private val itemId = savedStateHandle.get<String?>(NavigationConstants.ITEM_ID)
+        ?: savedStateHandle.get(NavigationConstants.TASK_ITEM_ID)
```

Comment:
Why would you sometimes have ITEM_ID and others TASK_ITEM_ID?

Reply:
It's some kind of legacy, currently, I can change this. (ITEM_ID was used for itemId that was passed through the agenda screen and TASK_ITEM_ID was used for navDeepLink)

Reply:
Ah
I would unify it into just one